simuleval/options.py

```diff
@@ -100,7 +100,12 @@ def add_evaluator_args(parser: argparse.ArgumentParser):
         default=-1,
         help="The last index for evaluation.",
     )
-    parser.add_argument("--output", type=str, default=None, help="Output directory")
+    parser.add_argument(
+        "--output",
+        type=str,
+        default=None,
+        help="Output directory. Required if using iterable dataloader.",
```

Comment:
Also just curious: why required if using iterable dataloader but not otherwise?

Reply:
For the typical dataloader (other than "iterable" ones), we create a list of instances and iterate through this list in the evaluator's `__call__` method, the results are stored as a filed in the instance object in this list and at the end of iteration we this list from memory to calculate the metrics/scores. So it doesn't matter if an output arg is specified, without an output arg the instances don't get written to a file but we will still be able to calculate the score/metrics using the results in memory.

But for iterable dataloaders where we are lazy loading the data one at a time, only the current instance is stored in memory and at each iteration we write this instance data to a file if `output` is given. At the end of iterating through all the data, we will load the results into memory using the file we wrote to while iterating and then use this in memory instances to calculate the cumulative scores/metric. So without the `output` arg, we won't write to file and subsequently we wont be able to calculate the score/metrics.